src/plugins/console/public/plugin.ts

```diff
@@ -97,5 +101,21 @@ export class ConsoleUIPlugin implements Plugin<void, void, AppSetupUIPluginDepen
     return {};
   }
 
-  public start() {}
+  public start(core: CoreStart, deps: AppStartUIPluginDependencies) {
+    const {
+      ui: { enabled: isConsoleUiEnabled },
+    } = this.ctx.config.get<ClientConfigType>();
+    if (isConsoleUiEnabled) {
+      return {
+        renderRemoteConsole: (props: RemoteConsoleProps) => {
```

Comment:
For now this seemed like a good way to expose the remote console vs exporting a component that other plugins could import from `@kbn/console-plugin/public` but I'm open to being wrong here. I'm not sure if there is precedent in other parts of Kibana for how to export a shared component like this.

Reply:
I haven't found many clear examples of this kind of thing. It's usually the other way around: something registers themselves elsewhere to be rendered there (eg: dev tools tabs). I like this pattern more, though.